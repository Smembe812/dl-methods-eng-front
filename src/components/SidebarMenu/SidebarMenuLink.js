import React from 'react';

import {
    NavLink,
    useRouteMatch
  } from "react-router-dom";

function SidebarMenuLink({ title, to, activeOnlyWhenExact, ...otherProps }) {
    let match = useRouteMatch({
      path: to,
      exact: activeOnlyWhenExact
    });
  
    return (
      <li className={match ? "toc__item toc__item--active" : "toc__item"}>
        <NavLink 
            className="toc__a" 
            activeClassName="toc__a--active" 
            to={to}
            {...otherProps}
        >
            {title}       
        </NavLink>
      </li>
    );
  }

export default SidebarMenuLink