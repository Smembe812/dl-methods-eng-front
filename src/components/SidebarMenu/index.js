import React from 'react';

import SidebarMenuLink from "./SidebarMenuLink"

function SidebarMenu({data}) {
    function renderListItems(data){

        const items = data.map((item, key) => {

            return  <SidebarMenuLink
                            key={key}
                            title={item}
                            to={`/${item.toLowerCase()}`}
                        />   
        })

        return items
    }

    if (data){
        return (
            <ul className="toc">
                {renderListItems(data)}
            </ul>
        );
    }

    return null

}

export default SidebarMenu