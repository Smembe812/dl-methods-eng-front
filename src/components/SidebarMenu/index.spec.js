import React from 'react'
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import { MemoryRouter } from "react-router-dom";
import renderer from 'react-test-renderer';

import SidebarMenu from './index'

describe('Render Sidebar', () => {
    let container = null;
    beforeEach((done) => {
      // setup a DOM element as a render target
      container = document.createElement("div");
      document.body.appendChild(container);
      done()
    });
    
    afterEach((done) => {
      // cleanup on exiting
      unmountComponentAtNode(container);
      container.remove();
      container = null;
      done()
    });

    it("should render", (done) => {
        act(() => {
            render(<SidebarMenu/>, container);
          });
      
          // const linkElement = getByText(/learn react/i);
          console.log(typeof container.textContent)
        expect(typeof container.textContent).toBe("string");
        done()
    })

    it("render sidebar", (done) => {
        const sidebarContent = [
            'Processes', 
            'Process Elements', 
            'Techniques', 
            'Experiences'
        ]

        const tree = renderer
        .create(
            <MemoryRouter initialEntries={['/']}>
                <SidebarMenu 
                    data={sidebarContent}/>
            </MemoryRouter>
        )
        .toJSON();
        expect(tree).toMatchSnapshot();
        done()
    })
});
