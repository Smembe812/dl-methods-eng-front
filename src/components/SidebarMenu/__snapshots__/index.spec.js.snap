// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Sidebar render sidebar 1`] = `
<ul
  className="toc"
>
  <li
    className="toc__item"
  >
    <a
      aria-current={null}
      className="toc__a"
      href="/processes"
      onClick={[Function]}
    >
      Processes
    </a>
  </li>
  <li
    className="toc__item"
  >
    <a
      aria-current={null}
      className="toc__a"
      href="/process elements"
      onClick={[Function]}
    >
      Process Elements
    </a>
  </li>
  <li
    className="toc__item"
  >
    <a
      aria-current={null}
      className="toc__a"
      href="/techniques"
      onClick={[Function]}
    >
      Techniques
    </a>
  </li>
  <li
    className="toc__item"
  >
    <a
      aria-current={null}
      className="toc__a"
      href="/experiences"
      onClick={[Function]}
    >
      Experiences
    </a>
  </li>
</ul>
`;
